aws_lambda_script/main.py

```diff
@@ -0,0 +1,189 @@
+import json
+import csv
+import os
+from typing import Dict, List
+import logging
+from io import StringIO
+import boto3
+
+# Configure logging
+logger = logging.getLogger()
+logger.setLevel(logging.INFO)
+
+# Environment variables
+SOURCE_BUCKET = os.environ.get("SOURCE_BUCKET", "sdp-dev-tech-audit-tool-api")
+SOURCE_KEY = os.environ.get("SOURCE_KEY", "new_project_data.json")
+DESTINATION_BUCKET = os.environ.get("DESTINATION_BUCKET", "sdp-dev-tech-radar")
+DESTINATION_KEY = os.environ.get("DESTINATION_KEY", "onsTechDataAdoption.csv")
+
+
+def get_json_from_s3(bucket: str, key: str) -> Dict:
+    """Retrieve JSON data from S3 bucket"""
+    try:
+        s3_client = boto3.client("s3")
+        response = s3_client.get_object(Bucket=bucket, Key=key)
+        json_data = json.loads(response["Body"].read().decode("utf-8"))
+        return json_data
+    except Exception as e:
+        logger.error("Error retrieving JSON from S3: %s", str(e))
+        raise
+
+
+def process_project_data(project_data: Dict) -> List[Dict]:
+    """Process project data into the required format"""
+    processed_data = []
+
+    projects = project_data.get("projects", [])
+    logger.info("Processing %d projects from JSON data", len(projects))
+
+    for project in projects:
+        row = {
```

Comment:
More of a nice to have than anything. It'd probably be a good idea to document the data format on both sides of the lambda. What JSON comes in and what CSV comes out? It'd be helpful to actually visualise what this is doing as it can be quite hard to read and get the idea.

Reply:
@evisos-ons I think you've misunderstood me here. I didn't mean to log all the ins and out, I meant to add an example data structure in your documentation.

i.e

# Example JSON
{
    "Project": "keh-tech-radar-lambda",
   "Project Short": "tech-radar-lambda"
}

# Example CSV
Project, Project Short, keh-tech-radar-lambda, tech-radar-lambda

Maybe in a markdown table?

| Project                            | Project Short         |
| -------------------------- | -------------------- |
| keh-tech-radar-lambda | tech-radar-lambda |

Reply:
Ah okay, thanks.

Reply:
Added to README.md